libs/community/langchain_community/llms/pieces.py

```diff
@@ -58,3 +50,33 @@ def _generate(
             text = self._call(prompt, stop=stop, run_manager=run_manager, **kwargs)
             generations.append([GenerationChunk(text=text)])
         return LLMResult(generations=generations)
+
+    def stream(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[GenerationChunk]:
+        """Stream the response from Pieces OS model."""
+        try:
+            for response in self.client.copilot.stream_question(prompt):
+                if response.question:
+                    answers = response.question.answers.iterable
+                    for answer in answers:
+                        yield GenerationChunk(text=answer.text)
+        except Exception as error:
+            print(f'Error streaming question: {error}')
+            yield GenerationChunk(text='Error streaming question')
+
+    def get_supported_models(self) -> List[str]:
+        """Get the list of supported models."""
+        return [model.value for model in ModelFoundationEnum]      
+
+    def set_model(self, model_name: str) -> None:
+        """Set the model to be used."""
+        if model_name in self.get_supported_models():
+            self.model = model_name
+            print(f"Model set to {model_name}. Note: This setting may not directly affect API calls.")
```

Comment:
Self.client.model_name = model_name

Reply:
You also don't need to verify that the model name is in the list I automatically do that and raise the same ValueError if it is not in the available_models_names list

Reply:
addressed in [36cd72c](https://github.com/shivay-at-pieces/langchain/pull/1/commits/36cd72ca483d74d119b57017b8690a8a663c626b) commit